### ISSUE TITLE
New version notifications

### ISSUE THREAD

**Issue by alidan** [deanoemcke/thegreatsuspender#733]
Not sure where to put this

Is it possible to add a feature that would push a new page, or push you to a page, or pop up a notification that won't go away till you read it?

When TGS updated, for me, it removed somewhere around 500 tabs, then it tried to bring them back for the better part of an hour, and I think it may be possible the extension got me a temporary ban with one of the sites I go to, a temporary one sure, but a ban nonetheless. 

With this in mind, would it be possible to push a notification, possibly a day possibly a week in advance so people can get themselves sorted out before an update comes? I have no real idea if it's possible to make an update non disruptive, but at least for me, if I know in advance I could use other extensions like tab outliner to mitigate the load an update will cause. 

Like I said, I have no idea where to put this

**Comment by alidan** [comment]
This is just my best guess as to what the ban is for as the site absolutely refuses to be specific.
Part of this was about the ban, the other part was the massive load that this had for the better part of an hour where I had to kill quite a few windows because they just were not helping chrome be responsive. a 1 day in advance notification and I could have closed out almost every single suspended tab so it would impact me less.

**Comment by fwextensions** [comment]
Looking at the code, it seems like tabs that had been suspended and were closed when the extension updated should have been reloaded in the suspended state.  Sounds like that didn't happen in your case, though.

**Comment by liamjohnston** [comment]
@fwextensions Yup, that's exactly what should happen - tabs that were suspended before update should still be suspended after the update (after they disappear and come back).

**Comment by deanoemcke** [comment]
@alidan this is something i have spent considerable time researching.
in the release you have just received i have implemented exactly what you have requested. unfortunately, the version you were upgrading _from_ did not have this code already.

in future (in best case scenario) the extension will warn you before it updates, and prompt you to back up your tabs in preparation.

due to the way the extension works, when it updates it kills all open tabs owned by that extension. this means every suspended tab. 
when the extension reloads, it looks for tabs that were suspended in the previous session and restores them in a SUSPENDED state. 
under no circumstances does the extension open a tab to an unsuspended url. this sounds like something that chrome would do on a chrome restart, when it tries to 'continue where you left off'.

from my experience, restoring even 500 suspended tabs should not take much more than 5 minutes. whereas 500 UNSUSPENDED tabs could well take an indefinite period of time. 
my best guess as to what happened to you (and i could well be wrong) is that you actually did have all these tabs unsuspended, and for some reason your chrome was restarted (possibly due to a crash caused by the extension update?), and when it restarted chrome tried to restore all your tabs as they were before the crash based on the 'continue from where you left off' policy.

let me know if this sounds feasible. if there's a bug in the extension update process i'd love to get to the bottom of it.

**Comment by alidan** [comment]
I was using chrome at the time that it did this, and several of the tabs are 'volatile' as in once closed they are gone for good, as none of those tabs are gone, I can safely say chrome did not restart. 

the problem with loading tabs suspended has been around for a while, when I do restart chrome, I have to load the program one window at a time otherwise everything will hang thankfully this usually means I load 1 window at a time, but in the case of the major steam sales where I will open windows in excess of 300-500 tabs, these can cripple everything for quite a while. 

I have no real idea what it did on update, but I did watch the program do about the same thing it does when I give it a massive amount of tabs to suspend. 

Looking forward to seeing the extension warn me.

**Comment by deanoemcke** [comment]
@alidan just curious, do you have the screencapture option turned on? this makes suspending tabs MUCH slower. if not, i've found it's pretty speedy at suspending large numbers of tabs. the new version queues up the tabs to be suspended so that it only tries to do at most 5 at a time to prevent huge CPU load. 

when you say 'the problem with loading tabs suspended has been around for a while', what do you mean exactly? when you 'load the program one window at a time', do you mean that you somehow restore windows from a session manager? and when you do, do those restored tabs get loaded as suspended or unsuspended? what session manager are you using to do this 'one window at a time' restoring?

**Comment by alidan** [comment]
No, Thought of haveing it turned on a while back but jpegs or png screenshots of every tab, which does get over 1000 at times, that just screams to me ram is going to be pissed away, but now with a 4k monitor and most of my windows being 1920x2160 the issue seems like it is going to get far FAR worse if I used that.

I should probably mention I am on windows 7, ryzen 1700 32gb of ram going at 3000mhz (or what ever number ryzen assigns it as it doesnt like 2800 or 3000 as ram speeds) with a 1060 6gb, my computer also stays on for the better part of inbetween power outs or blue screens (power outs are mitigated by a ups now, but blue screens due to bluetooth have been happening, I think I got the reason sorted though) so my computer has been on for 39 days as of right now, with chrome being shut down and restarted once in that time frame, this has caused a few issues to crop up, the most noticeable one being chrome demanding 3.2gb of vram, so trying to get to the point I can shut chrome down and restart it is a priority now.

generally on a fresh boot of chrome and after all the crap I want loaded is up, TGS is fairly fast, but over time, a suspend every tab on this page request can take up to half an hour to do, granted the fact I can do other things makes it an annoyance more than a problem, but the extension does slow down the longer chrome is used, or at least in my case, abused.
